src/System.Linq/src/System/Linq/Enumerable.cs

```diff
@@ -3567,6 +3559,11 @@ internal TElement[] ToArray()
         }
     }
 
+    // NOTE: DO NOT DELETE THE FOLLOWING DEBUG VIEW TYPES.
+    // Although it might be tempting due to them not be referenced anywhere in this library,
+    // Visual Studio currently depends on their existence to enable the "Results" view in 
+    // watch windows.
+
```

Comment:
I searched through VS/EE sources and nothing seem to be accessing _cachedCollection, or _count.
It seems they could as well be locals.

It kind of looks like _cachedCollection was intended to capture the results, but since it does not, I'd be careful to introduce caching now. If there is any caching in debug visualizers, it must be happening externally and adding caching here might be confusing.